templates/CRM/Profile/Form/Dynamic.tpl

```diff
@@ -218,12 +218,14 @@
       {/if}
       <div class="crm-submit-buttons" style='{$floatStyle}'>
         {include file="CRM/common/formButtons.tpl"}{if $isDuplicate}<span class="crm-button">{$form._qf_Edit_upload_duplicate.html}</span>{/if}
-        <a class="button cancel" href="{$cancelURL}">
-          <span>
-            <i class="crm-i fa-times"></i>
-            {$cancelButtonText}
-          </span>
-        </a>
+        {if $cancelButtonText}
```

Comment:
I understand the logic in buildQuickForm means there should be no $cancelButtonText unless the cancel button is enabled, but I feel like we should condition on the actual enable/disable option rather than a derivative.

Reply:
Done. New template variable is added `{$includeCancelButton}` to check if the cancel button is enabled on the form. Thanks for the comments @lcdservices